Makefile.pre.in

```diff
@@ -611,6 +611,20 @@ all:		@DEF_MAKE_ALL_RULE@
 # all.
 .PHONY: all
 
+# Provide quick help for common Makefile targets.
+.PHONY: help
+help:
+	@echo "Please use \`make <target>' where <target> is one of"
+	@echo "  clean      to remove build files"
+	@echo "  clinic     run Argument Clinic over source files"
+	@echo "  distclean  to remove other generated files (patch, exe, etc)"
```

Comment:
I think distclean should be just after clean, so that «other» makes sense.

In general I think sorting by usefulness (most frequent top) or phase (build targers, test targets, clean targets) would be more interesting than alphabetical order.

Reply:
Hmm... What's useful for you might not be for me. Do we arm wrestle to decide the best order? Alphabetical is at least consistent. Besides, I haven't included very many targets, so nothing is to y far from the head of the list.

Reply:
I don’t have a strong feeling, and no desire to wrestle with you on this 🙂

Reply:
moved distclean target